cmd/aad-cli/cli/user_test.go

```diff
@@ -0,0 +1,200 @@
+		"get user":                       {username: "myuser@domain.com"},
+		"get user, shadow not available": {username: "myuser@domain.com", shadowNotAvailable: true},
+
+		"get login":            {username: "myuser@domain.com", attribute: "login"},
+		"get password":         {username: "myuser@domain.com", attribute: "password"},
+		"get uid":              {username: "myuser@domain.com", attribute: "uid"},
+		"get gid":              {username: "myuser@domain.com", attribute: "gid"},
+		"get gecos":            {username: "myuser@domain.com", attribute: "gecos"},
+		"get home":             {username: "myuser@domain.com", attribute: "home"},
+		"get shell":            {username: "myuser@domain.com", attribute: "shell"},
+		"get last_online_auth": {username: "myuser@domain.com", attribute: "last_online_auth"},
+		"get shadow_password":  {username: "myuser@domain.com", attribute: "shadow_password"},
+
+		// error cases
+		"get nonexistent user":                      {username: "nouser@domain.com", wantErr: true},
+		"get bad_attribute":                         {username: "myuser@domain.com", attribute: "bad_attribute", wantErr: true},
+		"get shadow_password, shadow not available": {username: "myuser@domain.com", attribute: "shadow_password", shadowNotAvailable: true, wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			args := []string{"user"}
+			if tc.username != "" {
+				args = append(args, "--name", tc.username)
+			}
+			if tc.attribute != "" {
+				args = append(args, tc.attribute)
+			}
+			if tc.allUsers {
+				args = append(args, "--all")
+			}
+
+			cacheDir := t.TempDir()
+			testutils.CopyDBAndFixPermissions(t, filepath.Join("testdata", "cachedb"), cacheDir)
+			uid, gid := testutils.GetCurrentUIDGID(t)
+
+			shadowMode := -1
+			if tc.shadowNotAvailable {
+				shadowMode = 0
+			}
+			cache, err := cache.New(context.Background(), cache.WithCacheDir(cacheDir), cache.WithRootUID(uid), cache.WithRootGID(gid), cache.WithShadowGID(gid), cache.WithShadowMode(shadowMode))
+			require.NoError(t, err, "Setup: failed to create cache")
+			c := cli.New(cli.WithCache(cache))
+
+			got, err := testutils.RunApp(t, c, args...)
+			if tc.wantErr {
+				require.Error(t, err, "expected command to return an error")
+				return
+			}
+			require.NoError(t, err, "expected command to succeed")
+
+			if tc.username != "" {
+				user, err := cache.GetUserByName(context.Background(), tc.username)
+				require.NoError(t, err, "Setup: failed to get user from cache")
+				// Timestamps get serialized as RFC3339 which includes timezone
+				// information.
+				// We replace this with the unix timestamp to make
+				// the comparison easier.
+				got = timestampToUnix(t, got, user.LastOnlineAuth)
+			}
+
+			want := testutils.SaveAndLoadFromGolden(t, got)
+			require.Equal(t, want, got, "expected output to match golden file")
+		})
+	}
+}
+
+func TestUserSet(t *testing.T) {
```

Comment:
and here, I would say "TestUserSetAttribute" (even if that doesn’t match fully the command line, if you feel both tests should be kept separated as not having enough in common, this is fine)